main.tf

```diff
@@ -1,3 +1,17 @@
-resource "random_pet" "template" {
-  length = var.length
+locals {
+  users = { for u in var.users : u.username => u }
+  roles = {
+    "standard"  = data.datadog_role.standard.id
+    "admin"     = data.datadog_role.admin.id
+    "read_only" = data.datadog_role.read_only.id
+  }
+}
+
+resource "datadog_user" "users" {
+  for_each             = local.users
+  disabled             = each.value.disabled
+  email                = each.value.email
+  name                 = each.value.name
```

Comment:
I think the idea with lifecycle_ignore on the name was that users can change their name in the console and it causes a recreate or drift in the resource. We shouldn't care about how they want to spell their names or how they change them so we lifecycle ignored it... I'm thinking we add that back and make that clear via a comment. Sorry to reverse!

Reply:
Added. Sorry, this has been a bit hectic 😬

Reply:
@gberenice hectic? We just very quickly iterated on a bunch of good changes on this PR. This was kinda how I want all PRs to go. Things are going to need to change. We should be bringing up all of our thoughts on what is good practice or what we expect and then having fast feedback loops on iterating on those changes. This was great, hats off to you on doing things quickly and responding correctly to feedback 💯

Reply:
@Gowiem thanks for the great feedback!